Src/DependencyCollector/Shared/Implementation/HttpDesktopDiagnosticSourceSubscriber.cs

```diff
@@ -47,7 +52,20 @@ public void OnNext(DiagnosticListener value)
             {
                 if (value.Name == "System.Net.Http.Desktop")
                 {
-                    this.sourceSubscription = value.Subscribe(this.parent, (Predicate<string>)null);
+                    this.sourceSubscription = value.Subscribe(
+                        this.parent, 
+                        (evnt, r, _) =>
+                        {
+                            if (r != null && evnt == "System.Net.Http.Desktop.HttpRequestOut")
+                            {
+                                // request is never null
+                                var request = (HttpWebRequest)r;
+                                return !this.applicationInsightsUrlFilter.IsApplicationInsightsUrl(request.RequestUri.ToString());
```

Comment:
@cijothomas told that perf tests also override TelemetryChannel.Endpoint so `IsApplicationInsightsUrl` works.

But anyway it seems `SdkInternalOperationsMonitor.IsEntered()` should give better performance than string comparison